src/Engine/Combat/StrikeParams.java

```diff
@@ -55,7 +55,7 @@ public static StrikeParams buildStrikeParams(
         ( null == attacker.weapon ) ? 0 : attacker.weapon.getDamage(defender),
         isCounter);
 
-    if( gameMap.getEnvironment(attacker.coord).weatherType == Weathers.SIROCCO && !attacker.CO.immuneToSand )
+    if( null != attacker.env && attacker.env.weatherType == Weathers.SIROCCO && !attacker.CO.immuneToSand )
```

Comment:
You've changed this is a few places. If the attacker/defender likely to be on an invalid tile or something?
Note I don't object; its cleaner to have all the info together, but the messages seem to indicate a potential bug.

Reply:
The new Wally code considers hypotheticals without necessarily providing full gamestate info to the combat calc.
It wasn't a problem before (and I didn't feel like trying to make sure every calc got the map populated in every unit context, blegh)

Reply:
Yeah, UnitContext has always given me the heebie jeebies for this reason. It's too... manual to make sure it has everything it needs.

Reply:
The intent of allowing you to provide subsets is to... allow you to provide subsets.
In my mind, it makes sense to allow e.g. combat calculation without having a whole concrete map to look at. It won't take complex stuff like CO zones into account, but that's not unfixable either.